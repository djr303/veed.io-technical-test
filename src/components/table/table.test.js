import React from "react";
import renderer from "react-test-renderer";
import Table from ".";
import { mount } from "enzyme";

describe("<Table />", () => {
  it("should render correctly when no items have been passed through to props", () => {
    const wrapper = renderer
      .create(
        <Table
          items={void 0}
          handleChange={() => {}}
          title="Test title"
          favorites={{}}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly when there are items but no favorites", () => {
    const items = [
      {
        id: 1,
        name: "test name 1",
        html_url: "http://www.testurl1.com",
        description: "Test description 1",
        stargazers_count: 100,
      },
      {
        id: 2,
        name: "test name 2",
        html_url: "http://www.testurl2.com",
        description: "Test description 2",
        stargazers_count: 200,
      },
      {
        id: 3,
        name: "test name 3",
        html_url: "http://www.testurl3.com",
        description: "Test description 3",
        stargazers_count: 300,
      },
    ];

    const wrapper = renderer
      .create(
        <Table
          items={items}
          handleChange={() => {}}
          title="Test title"
          favorites={{}}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly when there are items but and some favorites", () => {
    const items = [
      {
        id: 1,
        name: "test name 1",
        html_url: "http://www.testurl1.com",
        description: "Test description 1",
        stargazers_count: 100,
      },
      {
        id: 2,
        name: "test name 2",
        html_url: "http://www.testurl2.com",
        description: "Test description 2",
        stargazers_count: 200,
      },
      {
        id: 3,
        name: "test name 3",
        html_url: "http://www.testurl3.com",
        description: "Test description 3",
        stargazers_count: 300,
      },
    ];

    const favorites = { 1: true, 2: false };

    const wrapper = renderer
      .create(
        <Table
          items={items}
          handleChange={() => {}}
          title="Test title"
          favorites={favorites}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should fire the handleChange callback when the favorite star is clicked", () => {
    const items = [
      {
        id: 1,
        name: "test name 1",
        html_url: "http://www.testurl1.com",
        description: "Test description 1",
        stargazers_count: 100,
      },
      {
        id: 2,
        name: "test name 2",
        html_url: "http://www.testurl2.com",
        description: "Test description 2",
        stargazers_count: 200,
      },
      {
        id: 3,
        name: "test name 3",
        html_url: "http://www.testurl3.com",
        description: "Test description 3",
        stargazers_count: 300,
      },
    ];

    const handleChangeSpy = jest.fn();

    const element = mount(
      <Table handleChange={handleChangeSpy} items={items} favorites={{}} />
    );
    element.find('img[data-test=1]').simulate("click");
    expect(handleChangeSpy).toHaveBeenCalledWith(1);
  });
});
