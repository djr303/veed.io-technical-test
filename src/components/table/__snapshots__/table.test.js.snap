// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render correctly when no items have been passed through to props 1`] = `
Array [
  <h1
    className="table-title"
  >
    Test title
  </h1>,
  <table
    className="items"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Link
        </th>
        <th>
          Description
        </th>
        <th>
          Stars
        </th>
        <th>
          Favorited
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
  <p
    className="none"
  >
    <i>
      (None)
    </i>
  </p>,
]
`;

exports[`<Table /> should render correctly when there are items but and some favorites 1`] = `
Array [
  <h1
    className="table-title"
  >
    Test title
  </h1>,
  <table
    className="items"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Link
        </th>
        <th>
          Description
        </th>
        <th>
          Stars
        </th>
        <th>
          Favorited
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          test name 1
        </td>
        <td>
          <a
            href="http://www.testurl1.com"
            target="_blank"
          >
            http://www.testurl1.com
          </a>
        </td>
        <td>
          Test description 1
        </td>
        <td>
          100
        </td>
        <td>
          <img
            data-test={1}
            onClick={[Function]}
            src="/images/star-filled.svg"
          />
        </td>
      </tr>
      <tr>
        <td>
          test name 2
        </td>
        <td>
          <a
            href="http://www.testurl2.com"
            target="_blank"
          >
            http://www.testurl2.com
          </a>
        </td>
        <td>
          Test description 2
        </td>
        <td>
          200
        </td>
        <td>
          <img
            data-test={2}
            onClick={[Function]}
            src="/images/star-outlined.svg"
          />
        </td>
      </tr>
      <tr>
        <td>
          test name 3
        </td>
        <td>
          <a
            href="http://www.testurl3.com"
            target="_blank"
          >
            http://www.testurl3.com
          </a>
        </td>
        <td>
          Test description 3
        </td>
        <td>
          300
        </td>
        <td>
          <img
            data-test={3}
            onClick={[Function]}
            src="/images/star-outlined.svg"
          />
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`<Table /> should render correctly when there are items but no favorites 1`] = `
Array [
  <h1
    className="table-title"
  >
    Test title
  </h1>,
  <table
    className="items"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Link
        </th>
        <th>
          Description
        </th>
        <th>
          Stars
        </th>
        <th>
          Favorited
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          test name 1
        </td>
        <td>
          <a
            href="http://www.testurl1.com"
            target="_blank"
          >
            http://www.testurl1.com
          </a>
        </td>
        <td>
          Test description 1
        </td>
        <td>
          100
        </td>
        <td>
          <img
            data-test={1}
            onClick={[Function]}
            src="/images/star-outlined.svg"
          />
        </td>
      </tr>
      <tr>
        <td>
          test name 2
        </td>
        <td>
          <a
            href="http://www.testurl2.com"
            target="_blank"
          >
            http://www.testurl2.com
          </a>
        </td>
        <td>
          Test description 2
        </td>
        <td>
          200
        </td>
        <td>
          <img
            data-test={2}
            onClick={[Function]}
            src="/images/star-outlined.svg"
          />
        </td>
      </tr>
      <tr>
        <td>
          test name 3
        </td>
        <td>
          <a
            href="http://www.testurl3.com"
            target="_blank"
          >
            http://www.testurl3.com
          </a>
        </td>
        <td>
          Test description 3
        </td>
        <td>
          300
        </td>
        <td>
          <img
            data-test={3}
            onClick={[Function]}
            src="/images/star-outlined.svg"
          />
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
